src/EntityGraphQL/Compiler/EntityGraphQLQueryWalker.cs

```diff
@@ -374,22 +374,11 @@ protected override void VisitInlineFragment(InlineFragmentNode node, IGraphQLNod
                 if (type != null)
                 {
                     var fragParameter = Expression.Parameter(type.TypeDotnet, $"frag_{type.Name}");
-                    IGraphQLNode newContext = context switch
-                    {
-                        GraphQLListSelectionField graphQLListSelectionField => new GraphQLListSelectionField(graphQLListSelectionField!, fragParameter),
```

Comment:
Wasn't this added to support the interfaces?

Reading it again I assume it was not correctly expanding fragments in interfaces?

Reply:
it was, but at that point I hadn't understood how the expand function was working for other fragments.  The new way is a lot more inline with how you were originally doing it and felt much less like a hack than what I had going.